plugin/src/main/java/com/denizenscript/denizen/events/entity/EntityEntersBlockScriptEvent.java

```diff
@@ -0,0 +1,87 @@
+package com.denizenscript.denizen.events.entity;
+
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizen.objects.EntityTag;
+import com.denizenscript.denizen.objects.LocationTag;
+import com.denizenscript.denizen.objects.MaterialTag;
+import com.denizenscript.denizen.utilities.implementation.BukkitScriptEntryData;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.scripts.ScriptEntryData;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.EntityEnterBlockEvent;
+
+public class EntityEntersBlockScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // <entity> enters <block>
+    //
+    // @Group Entity
+    //
+    // @Location true
+    //
+    // @Cancellable true
+    //
+    // @Triggers when an entity enters, and is stored in a block (EG a bee enters a bee nest)
+    // Does not fire when a silverfish "enters" a stone block. Prefer <@link event <entity> changes <block> (into <block>)> for that.
+    //
+    // @Context
+    // <context.entity> returns the EntityTag.
+    // <context.location> returns the LocationTag of the block entered by the entity.
+    // <context.material> returns the MaterialTag of the block entered by the entity.
+    //
+    // @Player when the entity that entered the block is a player
+    //
+    // @NPC when the entity that entered the block is an NPC.
+    //
+    // -->
+
+    public EntityEntersBlockScriptEvent() {
+        registerCouldMatcher("<entity> enters <block>");
+    }
+
+    public EntityTag entity;
+    public MaterialTag material;
+    public LocationTag location;
+    public EntityEnterBlockEvent event;
+
+    @Override
+    public boolean matches(ScriptPath path) {
+        String target = path.eventArgLowerAt(0);
+        if (!entity.tryAdvancedMatcher(target)) {
+            return false;
+        }
+        if (!path.tryArgObject(2, material)) {
+            return false;
+        }
+        if (!runInCheck(path, location)) {
+            return false;
+        }
+        return super.matches(path);
+    }
+
+    @Override
+    public ScriptEntryData getScriptEntryData() {
+        return new BukkitScriptEntryData(entity);
+    }
+
+    @Override
+    public ObjectTag getContext(String name) {
+        switch (name) {
+            case "entity": return entity;

```

Comment:
`.getDenizenObject()`

Reply:
I've added this so if it is OK to ask; what is the difference between this and returning the EntityTag directly as other scripts I look at seem to do?

I see from this it identifies if it's a Player or NPC and returns as that tag; does this mean events returning EntityTag cannot use player properties when obtained from context?